src/commands/Configuration/configure/autorole.ts

```diff
@@ -1,52 +1,113 @@
+
+	const placeholderRow = new MessageActionRow().addComponents(
+		new MessageSelectMenu()
+			.setCustomId("placeholder")
+			.setPlaceholder(
+				config.autorole !== null
+					? await client.bulbutils.translate("config_autorole_placeholder", interaction.guild?.id, { role })
+					: await client.bulbutils.translate("config_autorole_placeholder_disabled", interaction.guild?.id, {}),
+			)
+			.setDisabled(true)
+			.addOptions([
+				{
+					label: "Placeholder",
+					value: "placeholder",
+				},
+			]),
+	);
+
+	const [back, change, disable, header] = [
+		await client.bulbutils.translate("config_button_back", interaction.guild?.id, {}),
+		await client.bulbutils.translate("config_autorole_button_change", interaction.guild?.id, {}),
+		await client.bulbutils.translate("config_autorole_button_disable", interaction.guild?.id, {}),
+		await client.bulbutils.translate("config_autorole_header", interaction.guild?.id, {}),
+	];
+
+	const buttonRow = new MessageActionRow().addComponents([
+		new MessageButton().setCustomId("back").setLabel(back).setStyle("DANGER"),
+		new MessageButton().setCustomId("change").setLabel(change).setStyle("SUCCESS"),
+		new MessageButton()
+			.setCustomId("disable")
+			.setLabel(disable)
+			.setStyle("PRIMARY")
+			.setDisabled(config.autorole === null),
+	]);
+
+	interaction.deferred
+		? await interaction.editReply({ content: header, components: [placeholderRow, buttonRow] })
+		: await interaction.update({ content: header, components: [placeholderRow, buttonRow] });
 
-		if (role !== undefined && role !== null) {
-			await databaseManager.setAutoRole(<Snowflake>context.guild?.id, role!!.id);
-			return context.channel.send(await this.client.bulbutils.translate("config_autorole_success", context.guild?.id, { role: role!!.name }));
-		} else {
-			await databaseManager.setAutoRole(<Snowflake>context.guild?.id, null);
-			return context.channel.send(await this.client.bulbutils.translate("config_autorole_disable", context.guild?.id, {}));
+	const filter = i => i.user.id === interaction.user.id;
+	const collector = interaction.channel?.createMessageComponentCollector({ filter, time: 60000, max: 1 });
+
+	collector?.on("collect", async (i: MessageComponentInteraction) => {
+		if (i.isButton()) {
+			switch (i.customId) {
+				case "back":
+					await require("./main").default(i, client);
+					break;
+				case "disable":
+					await databaseManager.setAutoRole(interaction.guild?.id as Snowflake, null);
+					await interaction.followUp({
+						content: await client.bulbutils.translate("config_autorole_disable", interaction.guild?.id, {}),
+						ephemeral: true,
+					});
+
+					await autorole(i, client);
+					break;
+				case "change":
+					buttonRow.components[0].setDisabled(true);
+					buttonRow.components[1].setDisabled(true);
+					buttonRow.components[2].setDisabled(true);
```

Comment:
```suggestion
					for (const c of buttonRow.components) {
						buttonRow.components[c].setDisabled(true);
					}
```

Reply:
Test your code before suggesting, please. For of returns the object not an index and the code won't even compile if you try to use your suggestion

Reply:
Yep, that is true. My bad.
Now it's working.